projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -94,10 +95,17 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
     return controls;
   }
 
-  private mapControls<MapValue, T extends ControlMap<FormInterface, MapValue>>(
-    mapControl?: (ctrl: Controls<FormInterface>[keyof FormInterface], key: keyof FormInterface) => MapValue,
-    filterControl: (ctrl: Controls<FormInterface>[keyof FormInterface]) => boolean = () => true,
-  ): T | null {
+  private mapControls<MapValue>(
+    mapControl: MapControlFunction<FormInterface, MapValue>,
+    filterControl: FilterControlFunction<FormInterface>,
+  ): Partial<ControlMap<FormInterface, MapValue>> | null;
+  private mapControls<MapValue>(
+    mapControl?: MapControlFunction<FormInterface, MapValue>,
+  ): ControlMap<FormInterface, MapValue> | null;
+  private mapControls<MapValue>(
+    mapControl?: MapControlFunction<FormInterface, MapValue>,
+    filterControl: FilterControlFunction<FormInterface> = () => true,
+  ): Partial<ControlMap<FormInterface, MapValue>> | null {
```

Comment:
Gotta love overloads - this removes the need to have to type anything as `Required` as it implicitly returns the correct value when the `filterControl` param is omitted - when filter is provided it is assumed to return `Partial`. This fixes all the problems you were having :)

Reply:
:ok_hand: nice one, thanks!